vortex-layout/src/layouts/dict/writer/mod.rs

```diff
@@ -0,0 +1,188 @@
+use bytes::Bytes;
+use vortex_array::arcref::ArcRef;
+use vortex_array::compute::slice;
+use vortex_array::vtable::EncodingVTable as _;
+use vortex_array::{Array, ArrayContext, ArrayRef, RkyvMetadata, SerializeMetadata};
+use vortex_btrblocks::BtrBlocksCompressor;
+use vortex_dict::DictEncoding;
+use vortex_dict::builders::{DictConstraints, DictEncoder, dict_encoder};
+use vortex_dtype::{DType, PType};
+use vortex_error::{VortexResult, vortex_bail, vortex_err};
+
+mod repeating;
+mod single;
+
+use crate::layouts::dict::DictLayout;
+use crate::{Layout, LayoutStrategy, LayoutVTableRef, LayoutWriter, LayoutWriterExt};
+
+#[derive(Clone)]
+pub struct DictLayoutOptions {
+    pub repeat: bool,
+    pub constraints: DictConstraints,
+}
+
+impl Default for DictLayoutOptions {
+    fn default() -> Self {
+        Self {
+            repeat: true,
+            constraints: DictConstraints {
+                max_bytes: 1024 * 1024,
+                max_len: u16::MAX as usize,
+            },
+        }
+    }
+}
+
+pub struct DictStrategy {
+    pub options: DictLayoutOptions,
+    pub child: ArcRef<dyn LayoutStrategy>,
```

Comment:
Can you name this codes, and add an extra one called "fallback" for when we don't use dict encoding at all?